### ISSUE TITLE
TODO: Add title

### ISSUE THREAD

**Issue by Tonymmm** [koalaman/shellcheck#2151]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2213, SC2220, SC2214, SC2221, SC2222, SC2193
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
## Example: ShellCheck can detect some higher level semantic problems

while getopts "nf:" param
do
  case "$param" in
    f) file="$OPTARG" ;;
    v) set -x ;;
  esac
done

case "$file" in
  *.gz) gzip -d "$file" ;;
  *.zip) unzip "$file" ;;
  *.tar.gz) tar xzf "$file" ;;
  *) echo "Unknown filetype" ;;
esac

if [[ "$(uname)" == "Linux" ]]
then
  echo "Using Linux"
fi

```

#### Here's what shellcheck currently says:
```
Line 6	SC2213: getopts specified -n, but it's not handled by this 'case'.
Line 6	SC2220: Invalid flags are not handled. Add a *) case.
Line 8	SC2214: This case is not specified by getopts.
Line 13	SC2221: This pattern always overrides a later one on line 15.
Line 15	SC2222: This pattern never matches because of a previous pattern on line 13.
Line 19	SC2193: The arguments to this comparison can never be equal. Make sure your syntax is correct.
```
#### Here's what I wanted or expected to see:

TODO: Describe expected/desired output
